From faad0f29482c358e844f7e1e1d0ef0c38fdfa1ca Mon Sep 17 00:00:00 2001
From: Pankaj Bansal <pankaj.bansal@nxp.com>
Date: Wed, 29 Aug 2018 19:18:20 +0530
Subject: [PATCH 213/706] arm64: dts: freescale: add flexcan node in LS1028A

add device tree nodes for flexcan in LS1028A and enable the nodes
in LS1028A SOC based boards like RDB and QDS boards.

Signed-off-by: Pankaj Bansal <pankaj.bansal@nxp.com>
Signed-off-by: Zumeng Chen <zumeng.chen@windriver.com>
---
 .../boot/dts/freescale/fsl-ls1028a-qds.dts     |  8 ++++++++
 .../boot/dts/freescale/fsl-ls1028a-rdb.dts     |  8 ++++++++
 arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi | 18 ++++++++++++++++++
 3 files changed, 34 insertions(+)

diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
index 19a425e1bd82..0095475ccdd5 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-qds.dts
@@ -166,3 +166,11 @@
 		reg = <0>;
 	};
 };
+
+&can0 {
+	status = "okay";
+};
+
+&can1 {
+	status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
index 52900ee05587..1553787f7a69 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a-rdb.dts
@@ -152,3 +152,11 @@
 &duart1 {
 	status = "okay";
 };
+
+&can0 {
+        status = "okay";
+};
+
+&can1 {
+        status = "okay";
+};
diff --git a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
index bb7bc9b8a143..e6d3b86e99b4 100755
--- a/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
+++ b/arch/arm64/boot/dts/freescale/fsl-ls1028a.dtsi
@@ -500,6 +500,24 @@
 			#size-cells = <0>;
 			reg = <0x1 0xf8101c00 0x0 0x2c>;
 		};
+
+		can0: can@2180000 {
+			compatible = "fsl,ls1028ar1-flexcan", "fsl,lx2160ar1-flexcan";
+			reg = <0x0 0x2180000 0x0 0x10000>;
+			interrupts = <GIC_SPI 21 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clockgen 4 1>, <&clockgen 4 1>;
+			clock-names = "ipg", "per";
+			status = "disabled";
+		};
+
+		can1: can@2190000 {
+			compatible = "fsl,ls1028ar1-flexcan", "fsl,lx2160ar1-flexcan";
+			reg = <0x0 0x2190000 0x0 0x10000>;
+			interrupts = <GIC_SPI 22 IRQ_TYPE_LEVEL_HIGH>;
+			clocks = <&clockgen 4 1>, <&clockgen 4 1>;
+			clock-names = "ipg", "per";
+			status = "disabled";
+		};
 	};
 };
 
-- 
2.17.1

